can/notifier.py

```diff
@@ -51,31 +147,36 @@ def __init__(
         #: Exception raised in thread
         self.exception: Optional[Exception] = None
 
-        self._running = True
+        self._stopped = False
         self._lock = threading.Lock()
 
         self._readers: List[Union[int, threading.Thread]] = []
-        buses = self.bus if isinstance(self.bus, list) else [self.bus]
-        for each_bus in buses:
+        self._bus_list = self.bus if isinstance(self.bus, list) else [self.bus]
+        for each_bus in self._bus_list:
             self.add_bus(each_bus)
 
     def add_bus(self, bus: BusABC) -> None:
         """Add a bus for notification.
 
         :param bus:
             CAN bus instance.
+        :raises ValueError:
+            If the *bus* is already assigned to an active :class:`~can.Notifier`.
         """
-        reader: int = -1
+        # add bus to notifier registry
+        self._registry.register(bus, self)
```

Comment:
```suggestion
        Notifier._registry.register(bus, self)
```

Just to be clear that `_registry` not an instance attribute

Reply:
Done.